File: PDOO/Material Complementario/Tema7/Aerolinea/src/aerolinea/Vuelo.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package aerolinea;

import java.util.ArrayList;

/**
 *
 * @author pingu
 */
public class Vuelo {

    ArrayList<Viaje> viajes;
    Avion a;

    public Vuelo() {
        viajes = new ArrayList<>();
        a = null;
    }

    public void reservar(Pasajero pasajero) {
        boolean plazas = plazasLibres();
        if (plazas) {
            Asiento s = a.eligeAsientoLibre();
            Viaje elViaje = new Viaje(pasajero, s);
            if (plazas) { //Esto estaba mal en el diagrama
                viajes.add(elViaje);
            }
        }

    }
}

File: C1/PDOO/Material Complementario/Tema7/AerolineaRB/lib/vuelo.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Compania_aerea
  class Vuelo
    def initialize
      @viajes = Array.new
      @a = nil
    end

    def reservar(pasajero)
      plazas = plazas_libres
      if plazas
        s = a.elige_asiento_libre
        el_viaje = Viaje.new(pasajero,@a)
        if plazas # esto estaba mal en el diagrama
          @viajes << el_viaje
        end
      end
    end
  end
end
